pkg/registrymirror/registrymirror.go

```diff
@@ -69,6 +78,15 @@ func (r *RegistryMirror) CuratedPackagesMirror() string {
 	return r.NamespacedRegistryMap[constants.DefaultCuratedPackagesRegistryRegex]
 }
 
+// Credentials returns the credential for the registry mirror.
+func (r *RegistryMirror) Credentials() (string, string, error) {
```

Comment:
I'm not sure why we want this method with a `RegistryMirror` receiver. It doesn't depend on the state/dependencies of `RegistryMirror`.

By adding this wrapper, now we have two ways of doing the same thing and it's not obvious they are the same.

Moreover, IMO it increases the scope of `RegistryMirror` to something that is not very well defined. Before, it just wrapped around a registry mirror config and offered some helpers to read such config. How does reading envvars fit in that definition? What if we injected the credentials info when we construct it?